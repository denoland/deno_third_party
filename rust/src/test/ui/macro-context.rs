// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

// (typeof used because it's surprisingly hard to find an unparsed token after a stmt)
macro_rules! m {
    () => ( i ; typeof );   //~ ERROR expected expression, found reserved keyword `typeof`
                            //~| ERROR macro expansion ignores token `typeof`
                            //~| ERROR macro expansion ignores token `;`
                            //~| ERROR macro expansion ignores token `;`
}

fn main() {
    let a: m!();
    let i = m!();
    match 0 {
        m!() => {}
    }

    m!();
}
