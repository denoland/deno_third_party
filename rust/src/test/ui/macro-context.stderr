error: macro expansion ignores token `;` and any following
  --> $DIR/macro-context.rs:13:15
   |
LL |     () => ( i ; typeof );   //~ ERROR expected expression, found reserved keyword `typeof`
   |               ^
   |
note: caused by the macro expansion here; the usage of `m!` is likely invalid in type context
  --> $DIR/macro-context.rs:20:12
   |
LL |     let a: m!();
   |            ^^^^

error: macro expansion ignores token `typeof` and any following
  --> $DIR/macro-context.rs:13:17
   |
LL |     () => ( i ; typeof );   //~ ERROR expected expression, found reserved keyword `typeof`
   |                 ^^^^^^
   |
note: caused by the macro expansion here; the usage of `m!` is likely invalid in expression context
  --> $DIR/macro-context.rs:21:13
   |
LL |     let i = m!();
   |             ^^^^

error: macro expansion ignores token `;` and any following
  --> $DIR/macro-context.rs:13:15
   |
LL |     () => ( i ; typeof );   //~ ERROR expected expression, found reserved keyword `typeof`
   |               ^
   |
note: caused by the macro expansion here; the usage of `m!` is likely invalid in pattern context
  --> $DIR/macro-context.rs:23:9
   |
LL |         m!() => {}
   |         ^^^^

error: expected expression, found reserved keyword `typeof`
  --> $DIR/macro-context.rs:13:17
   |
LL |     () => ( i ; typeof );   //~ ERROR expected expression, found reserved keyword `typeof`
   |                 ^^^^^^ expected expression
...
LL |     m!();
   |     ----- in this macro invocation

error: aborting due to 4 previous errors

